Wrappers/Python/cil/framework/data_container.py

```diff
@@ -0,0 +1,1550 @@
+# -*- coding: utf-8 -*-
+#  Copyright 2024 United Kingdom Research and Innovation
+#  Copyright 2024 The University of Manchester
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+# Authors:
+# CIL Developers, listed at: https://github.com/TomographicImaging/CIL/blob/master/NOTICE.txt
+# Joshua DM Hellier (University of Manchester) [refactorer]
+
```

Comment:
```suggestion
# Nicholas Whyatt (UKRI-STFC) [refactorer]
```

I think it we should credit Nick @RedProkofiev here for his work on the refactoring during the hackathon

Reply:
@manchester-jhellier happy for me to add this?